### ISSUE TITLE
Running unregprod on an active/top 21 producer does not generate a new schedule

### ISSUE THREAD

**Issue by poplexity** [Telos-Foundation/telos#69]
The producer will continue to be listed in the active schedule and continue to produce blocks, with no proposed or pending schedule generated.

**Comment by poplexity** [comment]
`caleos@telos-testnet:~/installs/testnet1$ ./teclos.sh get schedule
active schedule version 135
    Producer      Producer key
    ============= ==================
    blindblocart  TLOS8keGPt9h9FDZkgJqUpcqcQve2T7qXK1viKSXuELc248MTK6AHX
    bpeosindexio  TLOS8MqHHaoxqZoBYfqBaEWzjF6oooCpsGyVP6kXBvBewz6F172LTK
    csxcommunity  TLOS54W39JNMYQE2Hji7ALcnVnALeVenZ7GaKeZfS6qCD2ANndo4cR
    eosbarcelona  TLOS7u9oCYhFSHTu8uHSFuZsk6Xzovu1X34RhYnSdf3bBUAgvj3J4y
    eosmetaliobp  TLOS6F5fHa3eXJoS3XfKxU19DDRUsPNPQSb8EAXgJqfJRWqLb3aSd6
    eosukblocpro  TLOS6f7n8QnTknmicDg41G84fDpL6UgNQdTDsoFnEWcfr1K1xThKDZ
    eosvancouver  TLOS555g7FHuzz9cPeoqBucDHXKQXCHFc8bRrpio2j8xsAhsDPu3dm
    infinitybloc  TLOS7hLp3cVtk94FCJGdNcF4r1azGgsWTk4139gTseB2TZVqaXdv4J
    swedencornet  TLOS67o9d17gGj1pDUufDaonsrAkasY7c2fsTWx7uqMjsNCsdNkUBh
    telgoingos13  TLOS5LdYveqLqSZbeD5FNRTu3C9KprG9QLv8rPrqouGN7iQBets7LD
    telos1russia  TLOS5QeNPyBt4sujved25g6EZSshfMoS9xAfwPmbJjs851vNwRx5x9
    teloscantons  TLOS69ECgLTUWXk31ZSxrpno88MFqxXTN8JRMZXHGrMNWH53kqA1ZH
    telosdact4st  TLOS61gUQNawb2JVA2xeVdkm9RTmVVpD4MsAzNteK9AnBCcvzDwsod
    telosflorida  TLOS5azFCaXovekBaLXYX11it2Hnx7P2AzuyVEWrVbjgfL7cJeJTqq
    telosgreenbp  TLOS8dJ28fSahNe9qvT97sgL82Rq4QtqKRDkm2TYKCFDZbaALvXBFs
    telosmiamibp  TLOS6n75eJfm3UcVu1BAdsDokJECLAqv92MSaiZFYeVMUXEok2dqpc
    telosnewyork  TLOS8T9iMFMS51NwFNoTtct6SQU9K699219zwJ6qf5PBg2zQ2QmDyT
    telosunlimit  TLOS6edyCXakJR2fzwGBCLuQf1uskaNcMHryCeokvuH36JmTsWCQoc
    telosvoyager  TLOS89u4ExdPgNWLnbMkeEoLNxxnucNB1Jkom3GQWH7SD6BD1KSgUk
    tlos21zbptn1  TLOS5QDBx5b6jd7LhRBdmG8faMmB5g21hKHJddGsR7T8VAuiU1qCEp
    tlsvenezuela  TLOS8c8cjcA7DHkZenn7Xhu7cDfPBshKTijJbagVn7wne1epzQvUne

pending schedule empty

proposed schedule empty

caleos@telos-testnet:~/installs/testnet1$ ./teclos.sh system listproducers
Producer      Producer key                                              Url                                                         Scaled votes
telosflorida  TLOS5azFCaXovekBaLXYX11it2Hnx7P2AzuyVEWrVbjgfL7cJeJTqq                                                                0.0490
tlos21zbptn1  TLOS5QDBx5b6jd7LhRBdmG8faMmB5g21hKHJddGsR7T8VAuiU1qCEp    http://www.21zephyr.com                                     0.0489
eosbarcelona  TLOS7u9oCYhFSHTu8uHSFuZsk6Xzovu1X34RhYnSdf3bBUAgvj3J4y    http://eosbarcelona.com                                     0.0489
swedencornet  TLOS67o9d17gGj1pDUufDaonsrAkasY7c2fsTWx7uqMjsNCsdNkUBh    https://eossweden.eu/                                       0.0489
eosukblocpro  TLOS6f7n8QnTknmicDg41G84fDpL6UgNQdTDsoFnEWcfr1K1xThKDZ                                                                0.0489
telosmiamibp  TLOS6n75eJfm3UcVu1BAdsDokJECLAqv92MSaiZFYeVMUXEok2dqpc    https://eos.miami                                           0.0489
telosvoyager  TLOS89u4ExdPgNWLnbMkeEoLNxxnucNB1Jkom3GQWH7SD6BD1KSgUk                                                                0.0489
telosgreenbp  TLOS8dJ28fSahNe9qvT97sgL82Rq4QtqKRDkm2TYKCFDZbaALvXBFs                                                                0.0489
csxcommunity  TLOS54W39JNMYQE2Hji7ALcnVnALeVenZ7GaKeZfS6qCD2ANndo4cR                                                                0.0489
tlsvenezuela  TLOS8c8cjcA7DHkZenn7Xhu7cDfPBshKTijJbagVn7wne1epzQvUne                                                                0.0489
eosvancouver  TLOS555g7FHuzz9cPeoqBucDHXKQXCHFc8bRrpio2j8xsAhsDPu3dm                                                                0.0489
telosdact4st  TLOS61gUQNawb2JVA2xeVdkm9RTmVVpD4MsAzNteK9AnBCcvzDwsod    95.179.136.183                                              0.0489
telos1russia  TLOS5QeNPyBt4sujved25g6EZSshfMoS9xAfwPmbJjs851vNwRx5x9                                                                0.0489
blindblocart  TLOS8keGPt9h9FDZkgJqUpcqcQve2T7qXK1viKSXuELc248MTK6AHX    http://www.blindblocart.io                                  0.0489
infinitybloc  TLOS7hLp3cVtk94FCJGdNcF4r1azGgsWTk4139gTseB2TZVqaXdv4J                                                                0.0489
telosnewyork  TLOS8T9iMFMS51NwFNoTtct6SQU9K699219zwJ6qf5PBg2zQ2QmDyT                                                                0.0489
eosmetaliobp  TLOS6F5fHa3eXJoS3XfKxU19DDRUsPNPQSb8EAXgJqfJRWqLb3aSd6                                                                0.0489
teloscantons  TLOS69ECgLTUWXk31ZSxrpno88MFqxXTN8JRMZXHGrMNWH53kqA1ZH                                                                0.0489
telgoingos13  TLOS5LdYveqLqSZbeD5FNRTu3C9KprG9QLv8rPrqouGN7iQBets7LD                                                                0.0489
telosmadrid1  TLOS8NUTmeyqq8AeDyNQaLjPVSc4VZP5vYwMuLEbK93QNpESRQbHp2                                                                0.0489
bpeosindexio  TLOS8MqHHaoxqZoBYfqBaEWzjF6oooCpsGyVP6kXBvBewz6F172LTK                                                                0.0000
amplifiedtls  TLOS7HtWCGGQRyAoZJ55zXHAFNDLcxkQQZ543zgoukpFSgdGWqBQyB                                                                0.0000
caleosblocks  TLOS532RbsTy3BrtNFwtgRbaQDJiwxJuwP5TVD1p7bH7n5ErewvaeF    https://caleos.io                                           0.0000
cryptomooner  TLOS1111111111111111111111111111111114T1Anm               cryptomooner.net                                            0.0000
defianttelos  TLOS63TBhRD8YchjDCSKVtVfJXHg5AavQQLx8WFumxkznMQtADm2i7                                                                0.0000
intelosgents  TLOS7VNrf9k2jVgfp5RBdjeNtYvpWJB9NyNvT436mdgX7tQXJeR5sa                                                                0.0000
kainostechtx  TLOS7mHdjzzyv1bpecSephVuKUCrUy2dHvQdm4m6A3cE5cUXf7tEJE    https://kainostech.com                                      0.0000
oraclizeosbp  TLOS7woMn7PL5YhguCK6VPMqGZwp7UbpB6wYYmr66yHmeuoPUyWyPf                                                                0.0000
telosdact3st  TLOS1111111111111111111111111111111114T1Anm               185.92.221.42                                               0.0000
telosintasia  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
testaccount4  TLOS1111111111111111111111111111111114T1Anm               https://testurl.com/                                        0.0000
tlosmalaysia  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
eosgreeniobp  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
telasiachina  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
telosdacbp11  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
telosgoooooo  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
telosislands  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
telosinspace  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0000
zalmoxisdeus  TLOS1111111111111111111111111111111114T1Anm               http://eos.romania22.com                                    0.0000
goodblockio1  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0052
telosunlimit  TLOS1111111111111111111111111111111114T1Anm                                                                           0.0489`

**Comment by jbuice** [comment]
dev2.7 has produced this same problem, also run into by @poplexity.

This appears similar to the issues we had on testnet where regpodding, kicking or re-voting doesn't do anything until the rotation timer runs out.

In this case with just 5 prods, we have no standbys so rotations won't happen.

**Comment by ARoomWithABue** [comment]
This is a symptom of another issue that has been occurring off and on during development. This issue is primarily caused when the onblock function fails to execute.`onblock` is an action that exists in the eosio.system contract. Every time a node produces a valid block, the current producer node sends a transaction to the `eosio.system::onblock` action. This action is the only form of automation that exists for the system contract, it's where we propagate vote weights, calculate missed blocks, create snapshots for claim rewards, and lastly and most importantly, propose new producer schedules. Actions that would otherwise need to be called manually by producers or members. The current bug (the schedule not updating) is actually a side effect of another bug, where `onblock` fails to execute. This can happen for several reasons. Contracts (or wasm under the hood) don't support exception handling due to the fact that the WebAssembly team hasn't implemented it yet, so when a contract would have an normal exception it instead either throws a wasm_assertion or fails silently with no information. 

@eedsilva  and I designed a strategy to get additional exception information from the `onblock` action. Normally the onblock action doesn't appear in the ABI for the system contract which makes it difficult to call using tools like eosjs. So we added the onblock function to the ABI and removed the `require_auth(N(eosio))`. This allowed us to call onblock directly using `teclos push action`, which is where discovered that RAM emplacement for the payments table was being paid for by the producers. `blindblocart` didn't have enough ram to pay for their emplacement. So we changed the design to have `eosio` pay for the RAM instead. We believe this resolved the problem.
